### ISSUE TITLE
Inaccurate distance in history

### ISSUE THREAD

**Issue by jacekn** [jonasoreland/runnerup#348]
In history view for my workouts there is a distance mismatch between total distance at the top of the screen and rightmost value on the graph. It is 6617m vs 7.92km so the difference is quite big.

I am pretty sure that it's the graph that is wrong.

On top of that map view shows seven kilometer markers which is wrong (should be just 6).
 
I am using Nexus5.

The problem can bee seen below:
![screenshot_20151025-195848](https://cloud.githubusercontent.com/assets/8809467/10717554/cf56bdf2-7b53-11e5-8a48-4ba4f86892e3.png)

**Comment by nebmo** [comment]
You are correct, I see the same bug. Don't know when it has been introduced..

**Comment by renegat** [comment]
This issue seems to be related with [Jaggy GPS track?](https://github.com/jonasoreland/runnerup/issues/356)
On a 4022m track I have a 5km distance marker in the map and 5,02km in the graph.

**Comment by LogicParrot** [comment]
Note that your average pace is `5:40`, however, all points on the graph are below `5:40`. This is impossible, isn't it? I am experiencing the same issue. In addition to an excessively long X axis, it seems the Y axis has issues too. I am certain that total distance and the average pace are accurate. Only the graph representation is not.

It also seems to me that the map view has distance issues. The last distance point on the map shows the wrong distance (in this case,  7.92km).

**Comment by LogicParrot** [comment]
Also, it is possible that the graph is fine, and only the text labels are misplaced.

**Comment by moobyfr** [comment]
I got the same problem on my moto G4. but I tried the current version from git, in android studio, and the graphs are fine.

**Comment by gerhardol** [comment]
Incorrect distance in Graph
Tagged as bug, but I recall something about corrections here (pauses?)

**Comment by davidedelvento** [comment]
It could also be related with smoothing, can you try disabling smoothing in the settings?
